backtracepython/report_queue.py

```diff
@@ -41,12 +42,7 @@ def process(self, report, attachments):
             self.source_code_handler.collect(report)
         self.request_handler.send(report, attachments)
 
-    def __del__(self):
-        self.dispose()
-
-    def dispose(self):
-        # Put a sentinel value to stop the worker thread
-        self.active = False
+    def finish(self):
         self.report_queue.put_nowait(None)
         self.report_queue.join()
-        self.worker_thread.join()
+        self.worker_thread.join(timeout=self.exit_timeout)
```

Comment:
If `exit_timeout` is `None`, this will block until all events have gone through. Is this intentional? Shouldn't we just quit immediately?

Reply:
I know that `None` is default, and when it's `None`, `join()` blocks until thread ends. That's why I asked.

Reply:
the thread ends after we process all reports - we still escape the thread when someone passes a sentinal value to the worker. None allows to wait until all reports are processed